src/ex2/DomManager.js

```diff
@@ -0,0 +1,201 @@
+class DomManager {
+    constructor() {
+        this.task_input = document.querySelector("#todo_input");
+        this.task_container = document.querySelector("#todo_tasks_container");
+        this.add_task_button = document.querySelector("#add_to_do_button");
+        this.tasks_counter = document.querySelector("#task_counter");
+        this.clear_all_button = document.querySelector("#clear_all_tasks_button");
+        this.sort_by_name_button = document.querySelector("#sort_by_name");
+        this.pokemon_data_list = document.querySelector("#pokemons_names");
+    }
+
+    /**
+     * clears input value
+     */
+    clearInput() {
+        this.task_input.value = "";
+    }
+
+    /**
+    * Click call back function to add new task
+    */
+    AddNewTask(id, task_object, delete_call_back) {
+        // creates new task elements
+        const new_task = document.createElement("li");
+        const task_text = document.createElement("span");
+        const task_number = document.createElement("span");
+        const task_buttons_container = document.createElement("div");
+        const task_delete_button = document.createElement("button");
+        const task_complete_button = document.createElement("button");
+
+        // adds class to elements
+        new_task.classList.add("todo_task");
+        new_task.classList.add("animate__animated");
+        new_task.classList.add("animate__fadeIn");
+        task_text.classList.add("task_text");
+        task_number.classList.add("task_number");
+        task_buttons_container.classList.add("task_buttons");
+        task_delete_button.classList.add("delete_task_button");
+        task_delete_button.id = id;
+        task_complete_button.classList.add("complete_task_button");
+
+        // Delete empty state from task container
+        this.task_container.classList.remove("empty");
+        // appends elements to dom
+        this.task_container.appendChild(new_task);
+        task_buttons_container.append(task_delete_button, task_complete_button);
+        new_task.append(task_number, task_text);
+        // image
+        if (Number.isInteger(task_object.id)) {
+            new_task.appendChild(this.PokemonImageElement(task_object));
+            // adds text to task
+            task_text.innerHTML = `Catch ${task_object.name}`;
+        }
+        else if(task_object.id === "Error")
+            task_text.innerHTML = task_object.data;
+        else
+            task_text.innerHTML = task_object;// adds text to task
+        new_task.append(task_buttons_container);
+
+        // adds numbers to tasks
+        task_number.innerText = (this.task_container.children.length).toString().concat(")");
+        this.clearInput();
+
+        // adds icons to buttons
+        task_delete_button.innerHTML = '<i class="fa-solid fa-trash-can"></i>';
```

Comment:
Putting this HTML in a constant will make your code more maintainable and readable